python/ql/src/experimental/semmle/python/Concepts.qll

```diff
@@ -13,3 +13,32 @@ private import semmle.python.dataflow.new.DataFlow
 private import semmle.python.dataflow.new.RemoteFlowSources
 private import semmle.python.dataflow.new.TaintTracking
 private import experimental.semmle.python.Frameworks
+
+/** Provides classes for modeling Regular Expression-related APIs. */
+module RegexExecution {
+  /**
+   * A data-flow node that works with regular expressions.
+   *
+   * Extend this class to model new APIs. If you want to refine existing API models,
+   * extend `RegexExecution` instead.
+   */
+  abstract class Range extends DataFlow::Node {
+    abstract DataFlow::Node getRegexNode();
+    abstract Attribute getRegexMethod();
```

Comment:
This would give us the __actual__ `re` method used and could be appended to the query's `select` statement as a string or as an `Attribute` (as it is now) for the UI to take you to the method in a click.